packages/libs/error-stack/src/lib.rs

```diff
@@ -205,6 +202,56 @@
 //! [`Display`]: core::fmt::Display
 //! [`Debug`]: core::fmt::Debug
 //!
+//! ### Multiple Errors
+//!
+//! [`Report`] supports the combination and propagation of multiple errors natively.
+//! This is useful in cases like parallel processing where multiple errors might happen
+//! independently from each other, in these use-cases you are able to use the implementations of
+//! [`Extend`] and [`extend_one()`] and are able to propagate all errors instead of just a single
+//! one.
+//!
+//! ```rust
+//! # #![cfg_attr(not(feature = "std"), allow(dead_code, unused_variables, unused_imports))]
+//! # use std::{fs, path::Path};
+//! # use error_stack::{IntoReport, Report};
+//! # pub type Config = String;
+//!
+//! # #[cfg(all(not(miri), feature = "std"))] {
+//! fn parse_configs(paths: &[impl AsRef<Path>]) -> Result<Vec<Config>, Report<std::io::Error>> {
+//!     let mut configs = Vec::new();
+//!     let mut error: Option<Report<std::io::Error>> = None;
+//!
+//!     for path in paths {
+//!         let path = path.as_ref();
+//!
+//!         match fs::read_to_string(path).into_report() {
+//!             Ok(ok) => {
+//!                 configs.push(ok);
+//!             }
+//!             Err(err) => {
+//!                 if let Some(error) = error.as_mut() {
+//!                     error.extend_one(err);
+//!                 } else {
+//!                     error = Some(err);
+//!                 }
+//!             }
+//!         }
+//!     }
+//!
+//!     if let Some(error) = error {
+//!         return Err(error);
+//!     }
+//!
+//!     Ok(configs)
+//! }
+//!
+//! # let report = parse_configs(&["test.txt", "test2.txt", "test3.txt"]).unwrap_err();
+//! # assert!(report.contains::<std::io::Error>());
+//! # }
+//! ```
```

Comment:
We don't run `miri` in doc tests anymore. Also, the doc tests are run with `--all-features`. This should simplify the examples a lot. I didn't adjust the other examples yet, but newly added examples can ignore the `cfg` dance.

Reply:
True, I created these example before merging #777 and didn't adjust them yet.